community/concurrent/pom.xml

```diff
@@ -14,7 +14,7 @@
   </properties>
 
   <modelVersion>4.0.0</modelVersion>
-  <artifactId>neo4j-primitive-collections</artifactId>
+  <artifactId>neo4j-concurrent</artifactId>
```

Comment:
All other primitive collections related comments/names in this pom should be updated as well. Also maybe we can clean up some module dependencies